### PR TITLE
Introduce `dragging` flag for MapBrowserEvent

### PR DESCRIPTION
**Pull request by tsauerwein**
So far, when listening to `mousemove`, there is no way to know if the map is currently being dragged. For example in the code below, `hasFeatureAtPixel` is always executed. Also when the map is being dragged, which slows down the performance.

```js
    $(map.getViewport()).on('mousemove', function(evt) {
      var pixel = map.getEventPixel(evt.originalEvent);
      var hit = map.hasFeatureAtPixel(pixel);
      map.getTarget().style.cursor = hit ? 'pointer' : '';
    });
```

This PR adds a flag `dragging` to `MapBrowserEvent`. With this, the above code can be changed to:

```js
    map.on('pointermove', function(evt) {
      if (evt.dragging) return;

      var pixel = map.getEventPixel(evt.originalEvent);
      var hit = map.hasFeatureAtPixel(pixel);
      map.getTarget().style.cursor = hit ? 'pointer' : '';
    });
```

### CODE REVIEW

**Comment by elemoine** [comment]
This looks great. I'm just wondering if `dragging` would be a better name.

**Comment by fredj** [comment]
FYI, I've already had to do that and I simply used the `which` property ot the event:
```javascript
$(map.getViewport()).on('mousemove', function(evt) {
  if (evt.originalEvent.which === 0) {
    // no button pressed
  }
});
```

**Comment by elemoine** [comment]
Not bad either. But I think a `dragging` flag would make a more robust solution.

**Comment by fredj** [comment]
`evt.originalEvent.which` is buggy on firefox (https://bugzilla.mozilla.org/show_bug.cgi?id=1048294).

File: src/ol/mapbrowserevent.js [hunk 1]
```diff
@@ -30,9 +30,11 @@ goog.require('ol.pointer.PointerEventHandler');
  * @param {string} type Event type.
  * @param {ol.Map} map Map.
  * @param {goog.events.BrowserEvent} browserEvent Browser event.
+ * @param {boolean} dragged Is the map currently being dragged?
  * @param {?olx.FrameState=} opt_frameState Frame state.
  */
-ol.MapBrowserEvent = function(type, map, browserEvent, opt_frameState) {
+ol.MapBrowserEvent = function(type, map, browserEvent, dragged,
```
Comment (fredj) [inline src/ol/mapbrowserevent.js]: What about changing `dragged` to an optional (`false` by default) parameter?
The `dragged` parameter should only be passed for `ol.MapBrowserEvent.EventType.POINTERDRAG` events: the value is false (by definition) for all the other events.

Comment (tsauerwein) [inline src/ol/mapbrowserevent.js]: Yes, good idea.

**Comment by tsauerwein** [comment]
> I'm just wondering if dragging would be a better name.

I also prefer `dragging`. I updated the branch, and it no longer breaks the `click` event.

**Comment by elemoine** [comment]
IIRC one of the editing interactions rely on the `down_` property (via the `getDown` method). We need to verify that your changes do not break that interaction. 

Otherwise looks perfect to me.

**Comment by tsauerwein** [comment]
> IIRC one of the editing interactions rely on the down_ property (via the getDown method). We need to verify that your changes do not break that interaction. 

`getDown()` [seems to be](https://github.com/openlayers/ol3/search?q=getdown&type=Code&utf8=%E2%9C%93) only used in a test.

**Comment by elemoine** [comment]
Hmm maybe that interaction's code changed.

File: src/ol/mapbrowserevent.js [hunk 2]
```diff
@@ -253,21 +264,24 @@ ol.MapBrowserEventHandler.prototype.emulateClickLegacyIE_ =
 ol.MapBrowserEventHandler.prototype.emulateClick_ = function(pointerEvent) {
   var newEvent;
   newEvent = new ol.MapBrowserPointerEvent(
-      ol.MapBrowserEvent.EventType.CLICK, this.map_, pointerEvent);
+      ol.MapBrowserEvent.EventType.CLICK, this.map_, pointerEvent,
+      this.dragging_);
```
Comment (fredj) [inline src/ol/mapbrowserevent.js]: `this.dragging_` can be omitted (same for all the other events except `ol.MapBrowserEvent.EventType.POINTERDRAG`)

**Comment by elemoine** [comment]
https://github.com/openlayers/ol3/commit/d6880039cc88102d26b2bf324ff49aa1f5311472 is the commit that removed the use of `getDown` from the Draw interaction. I think the `getDown` method should be removed entirely, and `this.down_` should be used instead in the tests. Could you please do that as part of this PR?

File: src/ol/mapbrowserevent.js (hunk 2 above)
Comment (tsauerwein) [inline src/ol/mapbrowserevent.js]: `dragging` is only set for `POINTERDRAG` and `POINTERMOVE`? Sounds reasonable. I will change it.

**Comment by tsauerwein** [comment]
> I think the getDown method should be removed entirely, and this.down_ should be used instead in the tests.

Yes, agree. Updated the branch.

File: src/ol/mapbrowserevent.js [hunk 3]
```diff
@@ -61,6 +63,12 @@ ol.MapBrowserEvent = function(type, map, browserEvent, opt_frameState) {
    */
   this.coordinate = map.getCoordinateFromPixel(this.pixel);
 
+  /**
+   * @type {boolean}
+   * @api stable
```
Comment (elemoine) [inline src/ol/mapbrowserevent.js]: The event should be documented.

**Comment by elemoine** [comment]
I've just added one last comment regarding documenting the new `dragging` event property. Please merge when this is done.

**Comment by tsauerwein** [comment]
Thanks for all your comments!

**Pull request merged**